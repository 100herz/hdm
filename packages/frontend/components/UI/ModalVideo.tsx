import styled from '@emotion/styled'

import Modal from '@/components/UI/Modal'

interface ModalVideoProps extends React.HTMLAttributes<HTMLElement> {
  videoName: string
  isOpen: boolean
  onClose?: () => void
  onVideoClick?: (event: React.MouseEvent<HTMLVideoElement, MouseEvent>) => void
}

export default function ModalVideo({ videoName, isOpen, onClose, onVideoClick, className = '' }: ModalVideoProps) {
  return (
    <Modal isOpen={isOpen} onClose={onClose} data-testid="modalVideo">
      <VideoContainer className={`${className} flex items-center justify-center m-auto`}>
        <video className="outline-none" controls autoPlay onClick={onVideoClick}>
          <source src={`/videos/${videoName}.webm`} type="video/webm" />
          <source src={`/videos/${videoName}.mp4`} type="video/mp4" />
        </video>
      </VideoContainer>
    </Modal>
  )
}

const VideoContainer = styled.div`
  @media (min-width: 768px) {
    max-width: 60%;
  }
`
