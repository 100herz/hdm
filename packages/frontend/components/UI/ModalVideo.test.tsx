import { render } from '@/test/test-utils'

import ModalVideo from './ModalVideo'

describe('ModalVideo component', () => {
  it('matches snapshot', () => {
    const { container, getByTestId } = render(<ModalVideo isOpen={true} videoName="" />)
    expect(getByTestId(/modalVideo/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})

// TODO: Test the functions
