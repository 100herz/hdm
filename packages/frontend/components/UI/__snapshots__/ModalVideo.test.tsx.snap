// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalVideo component matches snapshot 1`] = `null`;
